### PR TITLE
fix: add unauthenticated clone if authentication fails

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Add an attempt to run `git clone` if the authentication method cannot be determined

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

update testhelper to attempt an unauthenticated `git clone` if determining authentication method fails.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: testhelper/tests.go [hunk 1]
```diff
@@ -418,13 +407,45 @@ func (options *TestOptions) RunTestUpgrade() (*terraform.PlanStruct, error) {
...
-				// If unauthenticated clone also fails, return the error from the authenticated approach
-				return nil, fmt.Errorf("failed to clone base repo and branch with authentication: %v", errAuth)
+				// If unauthenticated clone fails and we cannot determine authentication, return the error from the unauthenticated approach
+				return nil, fmt.Errorf("failed to determine authentication method and clone base repo and branch without authentication: %v", errUnauth)
 			} else {
 				logger.Log(options.Testing, "Cloned base repo and branch without authentication")
 			}
 		} else {
-			logger.Log(options.Testing, "Cloned base repo and branch with authentication")
+			// Try to clone with authentication first
```
Comment (daniel-butler-irl) [inline testhelper/tests.go:416]: looks good, just update this comment and resolve conflicts.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**